firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1278,7 +1278,7 @@ class BrowserViewController: UIViewController,
 
     // MARK: - Microsurvey
     private func setupMicrosurvey() {
-        guard featureFlags.isFeatureEnabled(.microsurvey, checking: .buildOnly), microsurvey == nil else { return }
+        guard featureFlags.isFeatureEnabled(.microsurvey, checking: .userOnly), microsurvey == nil else { return }
```

Comment:
debating on making this a `.buildAndDebug` mode even if it relies on same logic as `userOnly`

Reply:
```
/// An enum representing the different types of checks we need to use for features.
/// All Nimbus default values are stored in the `nimbus.fml.yaml`
enum FlaggableFeatureCheckOptions {
    /// Checking only Nimbus value for the build.
    case buildOnly
    /// Checking Nimbus value first, then whether or not the user has set a preference,
    /// and, if no preference is set, the default Nimbus value.
    case buildAndUser
    /// Checking only the user preference. If no user preference exists,
    /// then nimbus defaults are used.
    case userOnly
}
```

These are the cases, by using build only, we may lose the value that a user set a preference or if we set the preference in the debug setting. Unless there is some context that I may be missing, let me know what you think!